interfaces/builtin/x11.go

```diff
@@ -134,6 +134,9 @@ owner /run/user/[0-9]*/.Xauthority r,
 owner /run/user/[0-9]*/.mutter-Xwaylandauth.* r,
 owner /run/user/[0-9]*/mutter/Xauthority r,
 
+# Allow reading KDE Plasma's Xwayland Xauth file
+owner /run/user/[0-9]*/xauth_* r,
```

Comment:
Bit of a shame they don't use `/run/user/[0-9]+/kwin/` or something similar for the path.

Reply:
It looks like this code is only a few weeks old, so it might be possible to convince them.  With that said, this isn't meaningfully different to the `$XDG_RUNTIME_DIR/.mutter-Xwaylandauth.*` files gnome-shell creates.

Reply:
should we add this to wayland.go too ,  for coherence? it has the .mutter entries already?

Reply:
I'm not entirely sure what the purpose of those rules are: Wayland does not make use of Xauthority authentication.

If you were running a snapped gnome-shell running as a nested compositor rendering to an X11 window, I'd expect you'd get these permissions as part of the x11 plug rather than the wayland slot.  If the intention was to let gnome-shell create the Xauthority file for the Xwayland it spawns, I'd expect the rule to be part of the x11 slot, and grant write access.

The x11 slot doesn't currently allow a snap to write to any of the common Xauthority locations, but that is independent of this issue.

Reply:
I talked about this with @oSoMoN, and he pointed me at https://github.com/snapcore/snapd/pull/7303#pullrequestreview-277883639.  I think it is based on a mis-reading of how the Xauthority file is handled by mutter/gnome-shell (and now KDE Plasma).

Rather than Xwayland writing out an xauth cookie that the Wayland compositor then reads, the compositor instead writes the xauth cookie itself before invoking Xwayland.

Reply:
so should we remove those rules from the wayland interface too?

Reply:
I don't think the rules make sense, so it would probably be worth removing them.  The number of snaps with a wayland slot is tiny, and the number containing a Mutter based compositor is even smaller.  So even if the rules did offer useful access, the impact would be tiny.

Reply:
I opened a PR here doing that: https://github.com/snapcore/snapd/pull/9969

Reply:
@zzag: I don't think there is an obviously better name than what you've chosen, and we would have needed changes on the snapd side to support it anyway (unless you used the same temporary file pattern as Mutter, which I wouldn't expect).

We've committed this change now, so changing the Xauthority file name again would require further changes.  As is, I think we should be fine to get this in a new snapd release before many more people encounter the problem.